File: app/src/main/java/com/example/calculatorfragment/JournalFragment.java
package com.example.calculatorfragment;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

public class JournalFragment extends Fragment {

    public JournalFragment() {
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_journal, container, false);
    }

     public void setSelectedItem(StringBuilder selectedItem) {
            TextView journal = getView().findViewById(R.id.journal);
            journal.setText(selectedItem);
     }

}

File: app/src/main/java/com/example/calculatorfragment/JournalFragment.kt
package com.example.calculatorfragment

import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.example.calculatorfragment.databinding.FragmentJournalBinding
import com.example.calculatorfragment.viewmodel.CalcViewModel

class JournalFragment : Fragment(R.layout.fragment_journal) {

    private lateinit var binding: FragmentJournalBinding
    private lateinit var viewModel: CalcViewModel

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {

        binding = DataBindingUtil.inflate(
            inflater,
            R.layout.fragment_journal,
            container,
            false
        )

        viewModel = ViewModelProvider(requireActivity()).get(CalcViewModel::class.java)

        binding.calcViewModel = viewModel

        binding.lifecycleOwner = this

        return binding.root
    }
}